File: go/1557.go
package main

func findSmallestSetOfVertices(n int, edges [][]int) []int {
	ret := make([]int, 0)
	in := make([]bool, n)

	for _, edge := range edges {
		in[edge[1]] = true
	}
	for i := range in {
		if !in[i] {
			ret = append(ret, i)
		}
	}

	return ret
}

File: cpp/1557.cpp
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> findSmallestSetOfVertices(int n, vector<vector<int>> &edges) {
    std::ios_base::sync_with_stdio(false);
    std::cout.tie(nullptr);
    std::cin.tie(nullptr);
    vector<bool> have_in(n, false);
    vector<int> ret;

    for (const auto &edge : edges)
      have_in[edge[1]] = true;

    for (int i = 0; i < n; i++)
      if (have_in[i] == 0)
        ret.push_back(i);

    return ret;
  }
};
